core/node/crypto/chain_txpool.go

```diff
@@ -246,8 +253,25 @@ func NewTransactionPoolWithPolicies(
 	return txPool, nil
 }
 
-func (tx *txPoolPendingTransaction) Wait() <-chan *types.Receipt {
-	return tx.listener
+func (tx *txPoolPendingTransaction) Wait(ctx context.Context) (*types.Receipt, error) {
```

Comment:
Would it also be useful to set error, if any, on the span?

Reply:
In this case the only error that can occur is the result of the ctx being cancelled/expiring. The reason this function returns an error is to force callers to deal with this situation and prevent nil panics when the receipt is nil if such a thing happens. Sop it won't be very interesting to set the error here.